tests/components/home_connect/test_coordinator.py

```diff
@@ -74,6 +79,54 @@ async def test_coordinator_update_failing_get_appliances(
     assert config_entry.state == ConfigEntryState.SETUP_RETRY
 
 
+@pytest.mark.usefixtures("setup_credentials")
+@pytest.mark.parametrize("platforms", [("binary_sensor",)])
+async def test_coordinator_update_fail_after_setup(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[MagicMock], Awaitable[bool]],
+    client: MagicMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test that the coordinator makes entities go unavailable on failure after setup."""
+    entity_id = "binary_sensor.washer_remote_control"
+    await async_setup_component(hass, "homeassistant", {})
+    await integration_setup(client)
+    assert config_entry.state == ConfigEntryState.LOADED
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state != "unavailable"
+
+    client.get_home_appliances.side_effect = HomeConnectError()
+
+    await hass.services.async_call(
+        "homeassistant", "update_entity", {"entity_id": entity_id}, blocking=True
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == "unavailable"
+
+    # Test that the entity becomes available again after a successful update
+
+    client.get_home_appliances.side_effect = None
+    client.get_home_appliances.return_value = copy.deepcopy(MOCK_APPLIANCES)
+
+    freezer.tick(timedelta(hours=1))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        "homeassistant", "update_entity", {"entity_id": entity_id}, blocking=True
+    )
```

Comment:
Why do we use the update_entity service here alongside freezing time?

Reply:
There's no polling by default. We have to do a forced poll via the service. There's still a debouncer on the coordinator that we need to pass.

Reply:
But why don't we use the push mechanism to trigger an update?

Reply:
I need to test both ways that data is received, by refresh and event stream. Here we're testing the coordinator refresh.